### PR TITLE
[FIX] _transform_plot_args was converting to degrees instead of coord_unit

### PR DESCRIPTION
**Pull request by nabobalis**
Discovered this when trying to plot a coordinate using a gWCS. 

I am unsure how to unit test this.

### CODE REVIEW

**Comment by Cadair** [comment]
Thanks @nabobalis !

Given you can't create a WCS with coord_type longitude and a coord_unit of anything other than deg with astropy I am not sure you can easily test this without importing gwcs into the tests.

If you could add a changelog that would be good though.

**Comment by nabobalis** [comment]
I added a simple changelog, let me if it's sufficient.

**Comment by WilliamJamieson** [comment]
> Thanks @nabobalis !
> 
> Given you can't create a WCS with coord_type longitude and a coord_unit of anything other than deg with astropy I am not sure you can easily test this without importing gwcs into the tests.
> 
> If you could add a changelog that would be good though.

@nabobalis would it be possible for you to contribute a test for this to GWCS?

**Comment by nabobalis** [comment]
> > Thanks @nabobalis !
> > Given you can't create a WCS with coord_type longitude and a coord_unit of anything other than deg with astropy I am not sure you can easily test this without importing gwcs into the tests.
> > If you could add a changelog that would be good though.
> 
> @nabobalis would it be possible for you to contribute a test for this to GWCS?

Don't see why not. Point me at a repo and I will try.

File: docs/changes/visualization/14251.bugfix.rst [hunk 1]
```diff
@@ -0,0 +1,2 @@
+``_transform_plot_args`` was converting to degrees instead of the unit of the coordinate frame.
+While not a problem for WCS, it is a problem for gWCS.
```
Comment (Cadair) [inline docs/changes/visualization/14251.bugfix.rst:2]: I would focus on the user facing affects of this:
```
``WCSAxes.plot_coord`` and ``plot_scatter`` now work correctly for APE 14 compliant WCSes where the units are not always converted to degrees.
```

**Comment by astrofrog** [comment]
> Given you can't create a WCS with coord_type longitude and a coord_unit of anything other than deg with astropy I am not sure you can easily test this without importing gwcs into the tests.

Couldn't we make a small fake APE 14 WCS to replicate this bug?

**Comment by Cadair** [comment]
> Couldn't we make a small fake APE 14 WCS to replicate this bug?

I am just thinking about what it would need to do, I am not sure how easy it would be to test the behaviour without an image test, so I think you would have to wrap a `astropy.wcs.WCS` object and do the unit conversion in `pixel_to_world_values`?

**Comment by WilliamJamieson** [comment]
> Don't see why not. Point me at a repo and I will try.

See https://github.com/spacetelescope/gwcs

File: docs/changes/visualization/14251.bugfix.rst (hunk 1 above)
Comment (nabobalis) [inline docs/changes/visualization/14251.bugfix.rst:2]:
```suggestion
``WCSAxes.plot_coord`` and ``plot_scatter`` now work correctly for APE 14 compliant WCSes where the units are not always converted to degrees.
```

**Comment by pllim** [comment]
We don't even declare gwcs to be installed in `setup.cfg`, so adding a GWCS object in test would be... controversial.

Is this good to go as is?

**Comment by nabobalis** [comment]
> Is this good to go as is?

As said I was told, but I imagine one of the WCSAXES maintainers needs to approve?

**Comment by pllim** [comment]
I guess depends on how badly @astrofrog wants you to make a fake WCS for a test here. Adding a test to GWCS downstream should be tracked over at GWCS repo separately.

Regardless, please squash the commits. Thanks!

**Comment by nabobalis** [comment]
> Regardless, please squash the commits. Thanks!

Done.

**Review by astrofrog — APPROVED** [review]
Looks good to me now!

**Comment by pllim** [comment]
I don't think we should auto backport this because we bumped the minversion of matplotlib on main but not the backport branches. That would mess with the hash stuff. If you really want to backport, please manually do it and use the approrpriate matplotlib versions being tested over there to generate the hashes.

**Comment by pllim** [comment]
Thanks, all!

**Comment by astrofrog** [comment]
I'm happy to backport manually but in future could you leave the backport labels if a PR is deemed to need backporting even if it will require a manual backport? Otherwise may lose track of it and also it's useful to have the bot instructions.

**Comment by astrofrog** [comment]
Thanks will backport later once at desk!

**Comment by nabobalis** [comment]
Thank you to @astrofrog and @pllim for getting this over the line.

**Pull request merged**